pkg/coscheduling/coscheduling.go

```diff
@@ -28,24 +28,38 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/klog"
-	"k8s.io/kubernetes/pkg/api/v1/pod"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
 )
 
 // Coscheduling is a plugin that implements the mechanism of gang scheduling.
+// TODO: implement a timeout based gc to clean up stale PodGroupInfo in the PodGroupInfos map.
 type Coscheduling struct {
 	frameworkHandle framework.FrameworkHandle
 	podLister       corelisters.PodLister
-	// Key is the name of Namespace/PodGroup.
+	// key is <namespace>/<PodGroup name> and value is *PodGroupInfo.
 	podGroupInfos sync.Map
 }
 
 // PodGroupInfo is a wrapper to a PodGroup with additional information.
-// TODO implement a timeout based gc for the PodGroupInfos map
+// A PodGroup's priority, temstamp and minAvailable are set according to
+// the values of the PodGroup's first pod that is added to the scheduling queue.
 type PodGroupInfo struct {
+	// key is a unique PodGroup ID and currently implemented as <namespace>/<PodGroup name>.
+	key string
+	// name is the PodGroup name and defined through a Pod label.
+	// The PodGroup name of a regular pod is empty.
 	name string
-	// timestamp stores the timestamp of the initialization time of PodGroup.
+	// isRegularPod indicates if the PodGroup is a regular Pod
+	isRegularPod bool
```

Comment:
Isn't it can be deduced by checking `key == ""`? I'm always in favor of having less private fields in a struct.

Reply:
removed isRegular and use len(pokey) == 0 instead.

Reply:
> BTW: @yuanchen8911 it's better to **not** amend the commit every time :) so that reviewers can quickly identify the delta changes.
Sorry about the inconvenience.  It seemed like  Permit was changed and merged after my PR submission. I had to rebase and merge the changes (squashed it again).